src/python_framework_v02/troute/nhd_network_utilities_v02.py

```diff
@@ -551,6 +553,122 @@ def build_channel_initial_state(
     return q0
 
 
+def build_forcing_sets(
+    forcing_parameters,
+    t0
+):
+    
+    run_sets = forcing_parameters.get("qlat_forcing_sets", None)
+    qlat_input_folder = forcing_parameters.get("qlat_input_folder", None)
+    nts = forcing_parameters.get("nts", None)
+    max_loop_size = forcing_parameters.get("max_loop_size", 12)
+    dt = forcing_parameters.get("dt", None)
+
+    try:
+        qlat_input_folder = pathlib.Path(qlat_input_folder)
+        assert qlat_input_folder.is_dir() == True
+    except TypeError:
+        print("Aborting simulation because no qlat_input_folder is specified in the forcing_parameters section of the .yaml control file.")
+        raise
+    except AssertionError:
+        print("Aborting simulation because the qlat_input_folder:", qlat_input_folder,"does not exist. Please check the the qlat_input_folder variable is correctly entered in the .yaml control file")
+        raise
+    
+    forcing_glob_filter = forcing_parameters.get("qlat_file_pattern_filter", "*.CHRTOUT_DOMAIN1")
+        
+    # TODO: Throw errors if insufficient input data are available
+
+    if run_sets:
+        
+        # append final_timestamp variable to each set_list
+        qlat_input_folder = pathlib.Path(qlat_input_folder)
+        for (s, _) in enumerate(run_sets):
+            final_chrtout = qlat_input_folder.joinpath(run_sets[s]['qlat_files'
+                    ][-1])
+            final_timestamp_str = nhd_io.get_param_str(final_chrtout,
+                    'model_output_valid_time')
+            run_sets[s]['final_timestamp'] = \
+                datetime.strptime(final_timestamp_str, '%Y-%m-%d_%H:%M:%S')
+            
+    else:
```

Comment:
It might be better to put an explicit `elif` here instead of just a bare `else`,  just for clarity.